src/modules/services_manager_service.rb

```diff
@@ -394,55 +350,39 @@ def exists?(service)
       end
     end
 
-    def switch_services
-      log.info "Switching services"
-      services_switched = []
-
-      services.each do |service_name, service_attributes|
-        next unless service_attributes[:modified]
-
-        service = Yast2::SystemService.find(service_name)
-        unless service
-          log.error "Cannot find service #{service_name}"
-          next
-        end
-
-        # Do not start or stop services that are already in the desired state.
-        # They might be coming from AutoYast import and thus marked as :modified.
-        if service.active? == service_attributes[:active]
-          log.info "Skipping service #{service_name} - it's already in desired state"
-        elsif switch!(service_name)
-          services_switched << service_name
-        else
-          change  = active(service_name) ? 'stop' : 'start'
-          status  = enabled(service_name) ? 'enabled' : 'disabled'
-          message = _("Could not %{change} %{service} which is currently %{status}. ") %
-            { :change => change, :service => service_name, :status => status }
-          message << status(service_name)
-          errors << message
-          Builtins.y2error("Error: %1", message)
-        end
+    # Returns a list of error messages for the given service
+    #
+    # @param service [String] Service name
+    # @return [Array<String>] List of error messages
+    def error_messages_for(service)
+      service.errors.keys.map do |key|
+        send("#{key}_error_message_for", service)
       end
-
-      services_switched
     end
 
-    def toggle_services
-      services_toggled = []
-      services.each do |service_name, service_attributes|
-        next unless service_attributes[:modified]
-        if set_start_mode!(service_name)
-          services_toggled << service_name
-        else
-          change  = enabled(service_name) ? 'enable' : 'disable'
-          message = _("Could not %{change} %{service}. ") %
-            { :change => change, :service => service_name }
-          message << status(service_name)
-          errors << message
-          Builtins.y2error("Error: %1", message)
-        end
-      end
-      services_toggled
+    # Returns a error message related to the activation/deactivation of services
+    #
+    # @return [String] Error message
+    def active_error_message_for(service)
+      change = service.active? ? 'start' : 'stop'
+      status = service.running? ? 'running' : 'not running'
+      _("Could not %{change} %{service} which is currently %{status}." %
+        { change: change, service: service.name, status: status })
```

Comment:
please, when you modify it, can you add some details, like why it failed?

Reply:
We can improve that in the future. I just wanted to keep the same error reporting that we had in the old version.